thundra/integrations/events/requests.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import absolute_import
+import traceback
+import thundra.constants as constants
+from urllib.parse import urlparse
+from thundra.integrations.base_integration import BaseIntegration
+
+class RequestsEvent(BaseIntegration):
+    def __init__(self, scope, wrapped, instance, args, kwargs, response, exception):
+        super(RequestsEvent, self).__init__()
+        
+        prepared_request = args[0]
```

Comment:
I would move the processing part to another method and call it from the factory. Currently it doesn't make sense to decorating span in the constructor